packages/markdown/remark/src/shiki.ts

```diff
@@ -57,6 +58,15 @@ export async function createShikiHighlighter({
 				transformers: [
 					{
 						pre(node) {
+							if (additionalProps) {
+								Object.entries(additionalProps).forEach(([key, value]) => {
+									if (key === 'class') {
+										node.properties[key] += ` ${value}`; // Append to class instead of replacing it
+									} else {
+										node.properties[key] = value as string;
+									}
```

Comment:
`class` and `style` should be handled specifically below using `classValue` and `styleValue`. The rest I think we can directly assign it.